enhancements/single-node-developer-cluster-profile.md

```diff
@@ -0,0 +1,514 @@
+profile, the existing SNC tests and CRC product tests will verify that
+nothing has changed in the resulting cluster.
+
+Once phase one is complete, a new OpenShift CI job will be added to
+run the Kubernetes and OpenShift conformance test suites on a cluster
+deployed using the single-node-developer profile. Tests that require
+components disabled in the profile will be skipped. Note that these
+conformance tests are run without the additional changes made by the
+SNC image build tooling, so they do not necessarily reflect the actual
+conformance of the CRC product.
+
+As part of this CI job, or as a separate job, the memory utilization
+of a single-node-developer cluster will also be checked, so that the
+impact of changes to OpenShift components, including changes made as
+part of phase two of this enhancement, can be tracked.
+
+The component testing required during the second phase will depend on
+what changes are made to each individual component. If a specialized
+manifest is added for a component, an appropriate subset of existing
+tests for that component should ideally be run using the new manifest
+(or the configuration values it specifies). If new configuration items
+or features are added to a component, those items will require new
+tests.
+
+As phase two proceeds, the set of conformance tests skipped in the
+single-node-developer CI job may also need to be adjusted. As phase
+two manifest modifications replace and eliminate changes currently
+made by the SNC image build tooling, these test results will more
+closely reflect the conformance of the actual CRC product.
+
+### Graduation Criteria
+
+This enhancement does not add or change any APIs or other user-visible
+features, so maturity levels and deprecation policies are not
+relevant.
+
+Graduation from phase 1 will happen when the SNC tooling is updated to
+use the new cluster profile. This should not result in any visible
+changes to users of the CRC product or to other users of SNC.
+
+As phase 2 is an open ended process of optimizing individual
+components, these changes will become available to SNC users,
+including CRC, as they are merged.
+
+### Upgrade / Downgrade Strategy
+
+The presence of the single-node-developer cluster profile, annotations
+for the profile on existing manifest files, and new specialized
+manifests annotated just for this profile will have no effect on
+clusters using any other cluster profile, including the default
+profile. Therefore, there should be no upgrade or downgrade concerns
+for any clusters not created via the SNC tooling.
+
+Clusters that are created via the SNC tooling are generally
+short-lived, and not upgradable across major OpenShift releases. No
+support is planned for upgrading or downgrading clusters to or from
+using the new profile.
+
+Upgrade / downgrade strategy may need to be addressed for new
+configuration items added to individual components to allow
+optimization for single node developer cluster usage.
```

Comment:
Not sure I follow, the previous paragraph clearly stated no upgrade/downgrade. Why any updates are needed then?

Reply:
If a component adds a feature to enable optimization for the single-node-developer profile, that feature might be used in other profiles that do support upgrade/downgrade. If so, then support might be needed for that new feature.

Reply:
i think i would say:

CRC does not support in-place upgrade/downgrade of a single-node developer cluster profile

the exception case @rkukura i think is understood that all other profiles of ocp need to keep functional behavior.